feat: added new nodes

YermekG opened the pull request:
* DX-2434: Added new nodes IsRegisteredOffchain and RegisteredOffchain
* DX-2514: Added new node Reconnect
* DX-2514: Removed redundant code
* fix: Fixed minor issue and added comments

nattb8 commented on Source/Immutable/Public/Immutable/ImmutablePassport.h:
```diff
@@ -503,6 +513,21 @@ class IMMUTABLE_API UImmutablePassport : public UObject {
+  void ImxIsRegisteredOffchain(const FImtblPassportResponseDelegate& ResponseDelegate);
+  
+  /**
+   * Registers a passport off-chain.
```
```suggestion
   * Register the user to Immutable X if they are not already registered
```

I just adapted it from [here](https://docs.immutable.com/docs/x/passport/imx-provider/#registeroffchain)

nattb8 commented on Source/Immutable/Public/Immutable/Actions/ImtblPassportHasStoredCredentialsAsyncAction.h:
Can we still keep this one? So it can be used to check if we should reconnect or not

YermekG replied:
Yes, I have reinstated and updated the `HasStoredCredentials `method.

suluvai reviewed (commented):
@Yermick, there are sample blueprints we ship along with the Unreal SDK. They will need to be updated to reflect the changes in this PR and also add functionality to test `RegisterOffchain` and `IsRegisteredOffChain` methods added in this repo.

suluvai reviewed (dismissed):
@Yermick, there are sample blueprints we ship along with the Unreal SDK. They will need to be updated to reflect the changes in this PR and also add functionality to test RegisterOffchain and IsRegisteredOffChain methods added in this repo.

YermekG commented:
> @Yermick, there are sample blueprints we ship along with the Unreal SDK. They will need to be updated to reflect the changes in this PR and also add functionality to test RegisterOffchain and IsRegisteredOffChain methods added in this repo.

I have updated the bridge UE asset via 4.26.2 version in order to guarantee its relevance across all UE versions. I added registerOffchain and isRegisterOffchain routine to the blueprint example files. I have tested the changed blueprint in both UE versions 4.26.2 and 5.0.3.

nattb8 commented on Source/Immutable/Private/Immutable/ImmutablePassport.cpp:
```diff
@@ -568,48 +545,46 @@ void UImmutablePassport::OnConnectPKCEResponse(FImtblJSResponse Response) {
-      ResponseDelegate->ExecuteIfBound(
-          FImmutablePassportResult{false, Msg, Response});
-    } else {
+    if (!Response.success || AccessToken.IsEmpty())
```
From testing, `success` can be true and `AccessToken` can be null. It's currently throwing an error when this happens. Please add this check to both Get Access Token and Get ID Token.

YermekG replied:
Ok, I have removed the empty token check and edited the log messages

nattb8 replied:
It still doesn't work. Also, I don't think you should remove that empty token check because it can still be empty.

It looks like the error is before `OnGetAccessTokenResponse` is called. It's complaining that `result` is not in the JSON object.

nattb8 commented on Source/Immutable/Private/Immutable/ImmutablePassport.cpp:
```diff
@@ -568,48 +545,42 @@ void UImmutablePassport::OnConnectPKCEResponse(FImtblJSResponse Response) {
+{
+  if (auto ResponseDelegate = GetResponseDelegate(Response))
+  {
+    if (!Response.success || !Response.JsonObject->HasField("result"))
```
I think we should also check that the `result` is not empty.
Same for OnGetIdTokenResponse and HasStoredCredentials

YermekG replied:
Fixed by using TryGet... methods
